#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_xenomai_include.dpatch by John Morris <john@zultron.com>
##
## Do a better job of detecting xeno-config.  Use the cflags returned
## by xeno-config in the kbuild Makefile.

@DPATCH@

diff -up ./configure.ac.xenomai_include ./configure.ac
--- ./configure.ac.xenomai_include	2013-02-12 10:31:08.000000000 -0600
+++ ./configure.ac	2013-08-30 15:32:10.000000000 -0500
@@ -491,37 +491,40 @@ AC_SUBST(RTAI_LXRT_LDFLAGS,[$rtai_lxrt_l
 # Xenomai path (optional)
 #------------------------------------------------------------------------------
 
-AC_ARG_WITH([xenomai-dir],
+xeno=0
+AC_ARG_WITH([xenomai],
     AC_HELP_STRING(
-        [--with-xenomai-dir=<DIR>],
-        [Xenomai path, for RTDM interface and Xenomai examples]
+        [--with-xenomai],
+        [Enable Xenomai]
     ),
     [
-        xenomaidir=[$withval]
-        xeno=1
-    ],
-    [
-        xenomaidir=""
-        xeno=0
+	test "$withval" = yes && xeno=1
     ]
 )
 
-AC_MSG_CHECKING([for Xenomai path])
-
-if test -z "${xenomaidir}"; then
-    AC_MSG_RESULT([not specified.])
-else
-    if test \! -r ${xenomaidir}/include/xeno_config.h; then
-        AC_MSG_ERROR([no Xenomai installation found in ${xenomaidir}!])
+if test $xeno = 1; then
+    xeno_config_path=''
+    AC_ARG_WITH([xeno-config],
+	AC_HELP_STRING(
+	    [--with-xeno-config=<path>],
+	    [The path to the 'xeno-config' executable]
+	),
+	[
+	    test $withval != no && xeno_config_path=":$withval"
+	]
+    )
+    
+    AC_PATH_PROG([xeno_config],[xeno-config],0,${PATH}${xeno_config_path})
+    if test $xeno_config = 0; then
+        AC_MSG_ERROR([Unable to find xeno-config executable])
     fi
-    AC_MSG_RESULT([$xenomaidir])
 
-    xeno_native_cflags=`$xenomaidir/bin/xeno-config --skin native --cflags`
-    xeno_native_ldflags=`$xenomaidir/bin/xeno-config --skin native --ldflags`
-    xeno_posix_cflags=`$xenomaidir/bin/xeno-config --skin posix --cflags`
-    xeno_posix_ldflags=`$xenomaidir/bin/xeno-config --skin posix --ldflags`
-    xeno_rtdm_cflags=`$xenomaidir/bin/xeno-config --skin rtdm --cflags`
-    xeno_rtdm_ldflags=`$xenomaidir/bin/xeno-config --skin rtdm --ldflags`
+    xeno_native_cflags=`$xeno_config --skin native --cflags`
+    xeno_native_ldflags=`$xeno_config --skin native --ldflags`
+    xeno_posix_cflags=`$xeno_config --skin posix --cflags`
+    xeno_posix_ldflags=`$xeno_config --skin posix --ldflags`
+    xeno_rtdm_cflags=`$xeno_config --skin rtdm --cflags`
+    xeno_rtdm_ldflags=`$xeno_config --skin rtdm --ldflags`
 fi
 
 AC_SUBST(XENOMAI_DIR,[$xenomaidir])
diff -up ./master/Kbuild.in.xenomai_include ./master/Kbuild.in
--- ./master/Kbuild.in.xenomai_include	2012-11-14 10:47:30.000000000 -0600
+++ ./master/Kbuild.in	2013-08-30 15:51:18.000000000 -0500
@@ -85,7 +85,7 @@ ifeq (@ENABLE_RTDM@,1)
 ec_master-objs += rtdm.o
 
 ifeq (@ENABLE_XENOMAI@, 1)
-CFLAGS_rtdm.o := -I@XENOMAI_DIR@/include
+CFLAGS_rtdm.o := @XENOMAI_RTDM_CFLAGS@
 endif
 
 ifeq (@ENABLE_RTAI@, 1)
